source/hsm.rst

```diff
@@ -0,0 +1,321 @@
+.. _doc_krill_hsm:
+
+Hardware Security Modules
+=========================
+
+Krill uses OpenSSL by default for key generation, and it stores the
+private keys thus generated in the `keys` directory under its data
+directory. For many organisations this will be good enough, provided
+of course that they ensure that access to the server and backup data
+is restricted.
+
+If you want to take security one (big) step up, then you may want to
+use a Hardware Security Module (HSM) instead. HSMs are devices that can
+be used to safeguard and manage digital keys. They are designed to allow
+applications to use private keys, whilst ensuring that the actual
+private keys are never leaked outside of the device - not even to the
+application that is using the key. The use of HSMs is supported in
+Krill 0.10.0-rc1 and up.
```

Comment:
Referring to RC1 is going to be outdated in a week. Why not just refer to Krill 0.10.0?

Reply:
Well... technically it won't be outdated - it would still be true albeit less relevant perhaps when this is tagged to the actual stable release for 0.10.0. On the other hand: saying 0.10.0 when it's not out yet could potentially be confusing to users wanting to test rc1.

I do see what you mean, but I was indeed planning to just update it when the 0.10.0 release happens.